modules/account-lib/src/coin/cspr/transactionBuilder.ts

```diff
@@ -305,12 +306,35 @@ export abstract class TransactionBuilder extends BaseTransactionBuilder {
     return session;
   }
 
+  /**
+   * Checks whether the transaction has the owner signature
+   * @param {string} pub - public key of the signer
+   * @returns {boolean} true if the pub key already signed th transaction
+   * @private
+   */
+  private isTransactionSignedByPub(pub: string): boolean {
+    return _.findIndex(this.transaction.casperTx.approvals, approval => {
+      const approvalSigner = approval.signer.slice(2).toUpperCase();
+      return approvalSigner === pub;
+    }) === -1;
```

Comment:
Shouldn't this be `!== -1`?
For what I can see, findIndex would return a number >= 0 iff `pub` has signed the transaction. So I think it should return true in that case.

Reply:
You're right, I had the logic inverted and the checks in transactionBuilder.ts too.
Fixed.